services/distribution/src/main/java/app/coronawarn/server/services/distribution/statistics/local/LocalStatisticsToProtobufMapping.java

```diff
@@ -0,0 +1,233 @@
+import app.coronawarn.server.services.distribution.statistics.exceptions.ConnectionException;
+import app.coronawarn.server.services.distribution.statistics.exceptions.FilePathNotFoundException;
+import app.coronawarn.server.services.distribution.statistics.file.JsonFile;
+import app.coronawarn.server.services.distribution.statistics.file.StatisticJsonFileLoader;
+import app.coronawarn.server.services.distribution.statistics.validation.StatisticsJsonValidator;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class LocalStatisticsToProtobufMapping {
+
+  private static final Logger logger = LoggerFactory.getLogger(LocalStatisticsToProtobufMapping.class);
+
+  private final StatisticJsonFileLoader jsonFileLoader;
+  private final LocalStatisticsDownloadService localStatisticsDownloadService;
+  private final RegionMappingConfig regionMappingConfig;
+
+  /**
+   * Process the JSON file provided by TSI and map the it to Local Statistics protobuf object.
+   *
+   * @param jsonFileLoader - Loader of the file from the system
+   * @param localStatisticsDownloadService - Local Statistics Download Service for keeping track of ETags
+   */
+  public LocalStatisticsToProtobufMapping(StatisticJsonFileLoader jsonFileLoader,
+      LocalStatisticsDownloadService localStatisticsDownloadService,
+      RegionMappingConfig regionMappingConfig) {
+    this.jsonFileLoader = jsonFileLoader;
+    this.localStatisticsDownloadService = localStatisticsDownloadService;
+    this.regionMappingConfig = regionMappingConfig;
+  }
+
+  /**
+   * Create protobuf local statistic map from raw JSON local statistics.
+   * Local statistics has to be uploaded on CDN in 7 packages. Each entry on the map represents 1 package.
+   * Packages of local statistics contains data about 1 up to 4 federal states and all their administrative units.
+   * For more info related to the grouping of the federal states into packages please look into
+   * https://github.com/corona-warn-app/cwa-app-tech-spec/blob/42e9e4f3c588cd2fd283f904e9f0ccd53a2b83d0/
+   * docs/spec/statistics.md#populate-local-statistical-data
+   *
+   * @return map containing local statistics grouped by archive id.
+   */
+  @Bean
+  public Map<Integer, LocalStatistics> constructProtobufLocalStatistics() {
+    Map<Integer, LocalStatistics> localStatisticsMap = new HashMap<>();
+
+    try {
+      Optional<JsonFile> file = this.getFile();
+
+      if (file.isEmpty()) {
```

Comment:
I think there might be an ifPresentOrElse that takes in a consumer and a runnable if the entry does not exist. This would also solve the issue with passing the optional as parameter. What do you think?

Reply:
Checking with a normal if gives me the possibility to return an empty map or the filled map. I don't know if I can return the bean value from a functional method. But I get rid of the optional because I can call optional.get() in the else condition. What do you think ?

Reply:
agree